tests/cupyx_tests/scipy_tests/sparse_tests/test_linalg.py

```diff
@@ -647,3 +647,97 @@ def test_mul(self, xp, sp):
         x1 = testing.shaped_random((self.N, 1), xp, self.dtype)
         x2 = testing.shaped_random((self.N, 8), xp, self.dtype)
         return linop * x0, linop * x1, linop * x2
+
+
+@testing.parameterize(*testing.product({
+    'lower': [True, False],
+    'unit_diagonal': [True, False],
+    'nrhs': [None, 1, 4],
+    'order': ['C', 'F']
+}))
+@unittest.skipUnless(scipy_available, 'requires scipy')
+@testing.gpu
+class TestSpsolveTriangular(unittest.TestCase):
+
+    n = 10
+    density = 0.5
+
+    def _make_matrix(self, dtype, xp):
+        a_shape = (self.n, self.n)
+        a = testing.shaped_random(a_shape, xp, dtype=dtype, scale=1)
+        mask = testing.shaped_random(a_shape, xp, dtype='f', scale=1)
+        a[mask > self.density] = 0
+        diag = xp.diag(xp.ones((self.n,), dtype=dtype))
+        a = a + diag
```

Comment:
We don't need the following step before this line?
```Python
if self.unit_diag:
    a[diag > 0] = 0
```

Reply:
No, that operation, `a[diag > 0] = 0`, is not necessary.

If `unit_diag == True`, you need to ignore the values stored in the diagonal part of the matrix and solve the linear equation with the diagonal value as `1.0`. Therefore, here the values stored in the diagonal part of the matrix are intentionally set to the values other than `1.0`.

https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.linalg.spsolve_triangular.html#scipy.sparse.linalg.spsolve_triangular

Reply:
`unit_diagonal` was added to `spsolve_triangular` in SciPy 1.4.0.
I didn't take that into account.
I'll add the requirement to the test.

Reply:
Okay, thanks! I had some misunderstanding.